redash/handlers/query_results.py

```diff
@@ -368,6 +368,8 @@ def get(self, query_id=None, query_result_id=None, filetype="json"):
                 response = self.make_json_response(query_result)
             elif filetype == "xlsx":
                 response = self.make_excel_response(query_result)
+            elif filetype == "tsv":
```

Comment:
Reaching 4 conditional branches, it might be time to refactor this into a simpler assignment in the style of `response = response_builders[filetype](query_result)`